specs/phase1/p2p-das.md

```diff
@@ -0,0 +1,244 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Ethereum 2.0 Phase 1 -- Network specification for Data Availability Sampling](#ethereum-20-phase-1----network-specification-for-data-availability-sampling)
+  - [Table of contents](#table-of-contents)
+  - [Introduction](#introduction)
+  - [DAS Subnets](#das-subnets)
+    - [Horizontal subnets](#horizontal-subnets)
+      - [Publishing](#publishing)
+      - [Horizontal propagation](#horizontal-propagation)
+      - [Horizontal to vertical](#horizontal-to-vertical)
+    - [Vertical subnets](#vertical-subnets)
+      - [Slow rotation: Backbone](#slow-rotation-backbone)
+      - [Quick Rotation: Sampling](#quick-rotation-sampling)
+  - [DAS in the Gossip domain: Push](#das-in-the-gossip-domain-push)
+    - [Topics and messages](#topics-and-messages)
+      - [Horizontal subnets: `shard_blob_{shard}`](#horizontal-subnets-shard_blob_shard)
+      - [Vertical subnets: `das_sample_{subnet_index}`](#vertical-subnets-das_sample_subnet_index)
+  - [DAS in the Req-Resp domain: Pull](#das-in-the-req-resp-domain-pull)
+    - [Messages](#messages)
+      - [DASQuery](#dasquery)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Ethereum 2.0 Phase 1 -- Network specification for Data Availability Sampling 
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+
+<!-- /TOC -->
+
+## Introduction
+
+For an introduction about DAS itself, see [the DAS section in the Phase 1 validator spec](./validator.md#data-availability-sampling).
+This is not a pre-requisite for the network layer, but will give you valuable context. 
+
+For sampling, all nodes need to query for `k` random samples each slot.
+
+*__TODO__: describe big picture of sampling workload size*
+
+This is a lot of work, and ideally happens at a low latency.
+
+To achieve quick querying, the query model is changed to *push* the samples to listeners instead, using GossipSub.
+The listeners then randomly rotate their subscriptions to keep queries unpredictable.
+Except for a small subset of subscriptions, which will function as a backbone to keep topics more stable.
+
+Publishing can utilize the fan-out functionality in GossipSub, and is easier to split between nodes:
+nodes on the horizontal networks can help by producing the same samples and fan-out publishing to their own peers.
+
+This push model also helps to obfuscate the original source of a message:
+the listeners will not have to make individual queries to some identified source.
+
+The push model does not aim to serve "historical" queries (anything older than the most recent).
+Historical queries are still required for the unhappy case, where messages are not pushed quick enough,
```

Comment:
Also in the case for historic queries

Reply:
> historical queries are still required... in case of historical queries?

I think I get what you mean, they are necessary for retrieving older missing data, but we should think of better wording before confusing others more.